src/app/shared/services/dialog-confirmation.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Rundown } from '../../core/models/rundown'
+import { RundownMode } from '../../core/enums/rundown-mode'
+import { BasicRundown } from '../../core/models/basic-rundown'
+import { RundownService } from '../../core/abstractions/rundown.service'
+import { BasicRundownStateService } from '../../core/services/basic-rundown-state.service'
+import { DialogService } from './dialog.service'
+import { Injectable } from '@angular/core'
+import { DialogSeverity } from '../components/confirmation-dialog/confirmation-dialog.component'
+
+@Injectable()
+export class DialogConfirmationService {
+  private readonly activationOkButtonText: string = $localize`dialog.button.activate`
+  private readonly activationMessage: string = $localize`dialog.message.activation`
+  private readonly activationWarningMessage: string = $localize`dialog.message.activation.warning`
+
+  private readonly rehearseOkButtonText: string = $localize`dialog.button.rehearse`
+  private readonly rehearseMessage: string = $localize`dialog.message.rehearse`
+  private readonly rehearseWarningMessage: string = $localize`dialog.message.rehearse.warning`
+
+  constructor(
+    private readonly dialogService: DialogService,
+    private readonly basicRundownStateService: BasicRundownStateService,
+    private readonly rundownService: RundownService
+  ) {}
+
+  public openActivateRundownDialog(rundown: Rundown): void {
```

Comment:
If i have a rundown in `REHEARSAL` and i then try to activate it (as "on air"), i will now be asked if that same rundown should be deactivated when i try to do so. I don't know if this is a huge issue, but it is for sure not the way it should be.

Reply:
Great catch.